### DISCUSSION TITLE
I2C configurations

### DISCUSSION THREAD

**Discussion by sorriso93** [RavenSystem/esp-homekit-devices#1727] [MEPLHAA Scripts Collection]
Does anyone have valid configuration with I2C sensors? 
I'm interested to use a SHT30 and BH1750. Ideally would be great also an example on SGP30 and HP303B.
Thanks

**Comment by TriderG75** [comment]
> Does anyone have valid configuration with I2C sensors?
> I'm interested to use a SHT30 and BH1750. Ideally would be great also an example on SGP30 and HP303B.
> Thanks

BH1750 already exists directly implemented into haa with "t":50 accessory. Do you need anyway the version for free monitor?

**Reply by pawelosin** [reply]
My json for light sensor BH1750 and data history service 
{"c": {"o": 0,"l": 2,"i": 1,"ic": [[4,5,400]],"n": "light-sensor","ntp": "pl.pool.ntp.org","tz": "CET-1CEST-2,M3.5.0/2,M10.5.0/3","ct": 10,"b": [{"g": 0}]},"a": [{ "t": 50,"n": 2,"dt":[0,35] ,"es": [{"t": 95,"h": [1, 0, 5]}]}]}

**Comment by Kienz** [comment]
Does anybody have an configuration for I2C SHT30 temp/hum. sensor?

**Reply by Kienz** [reply]
This is my JSON for SHT30 sensor connected to M5Stack Atom Lite ESP32 module.
```json
{"c":{"io":[[[27],3],[[39],6]],"l":27,"b":[{"g":39,"t":5}],"ic":[[32,26,100]],"ntp":"de.pool.ntp.org","tz":"CET-1CEST-2,M3.5.0/2,M10.5.0/3"},"a":[{"t":24,"es":[{"t":80,"tg":[1,0],"n":20,"j":1,"ic":[0,68],"in":[[0,33,48]],"dt":[2,0],"ff":0.002670328831922,"fo":-45},{"t":80,"tg":[1,1],"n":20,"j":1,"ic":[0,68],"dt":[2,0],"io":3,"ff":0.00152590218967}]}]}
```

**Reply by xrust83** [reply]
Help "ic": and "in": for HTU21D
I calculated the coefficients of temperature and humidity

<img width="1177" alt="Знімок екрана 2023-05-11 о 11 28 04" src="https://github.com/RavenSystem/esp-homekit-devices/assets/58448955/0884e7b1-6fd8-4533-aca4-39c24edc22be">

Temp
"ff": 0.002681274414063,
"fo": -46.85
Humidity
"ff":  0.001907348632813,
"fo": -6,
I think that "io":[0,64], but I don't understand how to write "in": 
https://cdn-shop.adafruit.com/datasheets/1899_HTU21D.pdf


{"c":{"io":[[[2],3],[[0],6]],"l":2,"b":[{"g":0,"t":5}],"ic":[[4,5,100]]},"a":[{"t":24,"es":[{"t":80,"tg":[1,0],"n":20,"j":1,"ic":[0,64],"in":[[?,?,?]],"dt":[2,0],"ff":0.002681274414063,"fo":-46.85},{"t":80,"tg":[1,1],"n":20,"j":1,"ic":[0,64],"dt":[2,0],"io":3,"ff":0.0019073448632813,"fo":-6}]}]}

**Reply by xrust83** [reply]
If I understand correctly, I need type "n":21 I2C trigger command, and "it": but I can't figure out the command view. I need to send 0x80, 0xE3, 0x81 (128, 227, 129), but I don't understand where. Can someone explain?

**Comment by Kienz** [comment]
Did you try "in":[[0,128,227,129]]?
```json
{"c":{"io":[[[2],3],[[0],6]],"l":2,"b":[{"g":0,"t":5}],"ic":[[4,5,100]]},"a":[{"t":24,"es":[{"t":80,"tg":[1,0],"n":20,"j":1,"ic":[0,64],"in":[[0,128,227,129]],"dt":[2,0],"ff":0.002681274414063,"fo":-46.85},{"t":80,"tg":[1,1],"n":20,"j":1,"ic":[0,64],"dt":[2,0],"io":3,"ff":0.0019073448632813,"fo":-6}]}]}
```

**Reply by xrust83** [reply]
Yes, of course, this is the easiest thing that could be tried. I think everything is more complicated, and the sensor is processed through a trigger https://github.com/RavenSystem/esp-homekit-devices/wiki/Free-Monitor#i2c-trigger-command
<img width="908" alt="Знімок екрана 2023-05-12 о 16 24 27" src="https://github.com/RavenSystem/esp-homekit-devices/assets/58448955/84ee8d40-19ac-4df7-b07b-a4944031f9f8">
<img width="900" alt="Знімок екрана 2023-05-12 о 16 24 55" src="https://github.com/RavenSystem/esp-homekit-devices/assets/58448955/cad4ae92-1eb2-4c5d-a215-ea447e9a6b59">
I do not understand how to read LBS and MBS
